### PR TITLE
Optimize changelogdata deduplication

### PR DESCRIPTION
**Pull request by witekest**
## What does this PR change?

[DO NOT MERGE]

Attempt to optimize cleaning up the duplicates in `rhnpackagechangelogdata` table.

## GUI diff

No difference.

Before:

After:

- [ ] **DONE**

## Documentation
- No documentation needed: **add explanation. This can't be used if there is a GUI diff**
- No documentation needed: only internal and user invisible changes
- Documentation issue was created: [Link for SUSE Manager contributors](https://github.com/SUSE/spacewalk/issues/new?template=ISSUE_TEMPLATE_DOCUMENTATION.md&labels=documentation&projects=SUSE/spacewalk/31), [Link for community contributors](https://github.com/uyuni-project/uyuni-docs/issues/new).
- API documentation added: please review the Wiki page [Writing Documentation for the API](https://github.com/uyuni-project/uyuni/wiki/Writing-documentation-for-the-API) if you have any changes to API documentation.
- (OPTIONAL) [Documentation PR](https://github.com/uyuni-project/uyuni-docs/pulls)

- [ ] **DONE**

## Test coverage
- No tests: **add explanation**
- No tests: already covered
- Unit tests were added
- Cucumber tests were added

- [ ] **DONE**

## Links

Issue(s): #
Ports(s): # **add downstream PR(s), if any**

- [ ] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

# Before you merge

Check [How to branch and merge properly](https://github.com/uyuni-project/uyuni/wiki/How-to-branch-and-merge-properly)!

### CODE REVIEW

**Comment by admd** [comment]
Can we also recheck and see if we can concurrently add the index like before but with "if not exists" clause? https://github.com/SUSE/spacewalk/commit/9c41645710b45a6afb6b8fdc84ff975db5533ce5

File: schema/spacewalk/upgrade/susemanager-schema-4.4.8-to-susemanager-schema-4.4.9/203-add-changelogdata-index.sql [hunk 1]
```diff
@@ -1,27 +1,28 @@
...
+
+        delete from rhnpackagechangelogdata
+        where id in (select unnest(duplicate_ids) from (
+            select
+                min(id) as keep_id,
+                array_agg(id) as duplicate_ids
+            from rhnpackagechangelogdata
+            group by name, text, time
+            having count(*) > 1
+        ) as d where d.keep_id <> any(d.duplicate_ids));
```
Comment (mcalmer) [inline schema/spacewalk/upgrade/susemanager-schema-4.4.8-to-susemanager-schema-4.4.9/203-add-changelogdata-index.sql:25]: Are we sure that this is not removing the keep_id as well?
This `<> ANY` is not clear to me what it exactly does.

Comment (witekest) [reply schema/spacewalk/upgrade/susemanager-schema-4.4.8-to-susemanager-schema-4.4.9/203-add-changelogdata-index.sql:25]: You're right. That's bogus. Fixing.

**Comment by mackdk** [comment]
From what I could test on my database dump of manager.mgr.suse.de, the update hits a duplicate constraint check:

```
ERROR:  duplicate key value violates unique constraint "rhn_pkg_clr_pid_cld_uq"
DETAIL:  Key (package_id, changelog_data_id)=(264816, 589384) already exists.
```

This is because the same package is linked with two changelog data, that are in fact duplicates: 

```
 package_id | changelog_data_id |          name          |                            text                            |          time          
------------+-------------------+------------------------+------------------------------------------------------------+------------------------
**   264816 |            589384 | a#@#######e.de         |                                                            | 2013-02-05 12:00:00+01 **
**   264816 |            589386 | a#@#######e.de         |                                                            | 2013-02-05 12:00:00+01 **
     264816 |            589385 | a#####@###e.cz         |                                                            | 2010-07-16 11:00:00+02
     264816 |            705689 | a#####@###e.cz         | - update to 0.97                                          +| 2007-10-08 11:00:00+02
            |                   |                        |   * Allow for better handling of setlogsock depending upon+| 
            |                   |                        |     the version of Sys::Syslog installed                  +| 
            |                   |                        |   * Update examples with minimal pod and working synopses +| 
            |                   |                        |   * Added post_client_connection_hook                      | 
     264816 |            589387 | c####@############t.de |                                                            | 2010-07-17 11:00:00+02
     264816 |            589383 | c####@############t.de |                                                            | 2010-07-20 11:00:00+02
     264816 |            496738 | c####@#####l.com       |                                                            | 2010-11-30 12:00:00+01
     264816 |            510994 | c####@###e.com         |                                                            | 2011-11-18 12:00:00+01
     264816 |            571718 | c####@###e.com         |                                                            | 2011-12-06 12:00:00+01
     264816 |            496739 | j######@######s.de     |                                                            | 2010-01-10 12:00:00+01
(10 rows)
```
In this case the row should be deleted, not updated. I'm not sure though how we got into this situation.

**Comment by witekest** [comment]
We have decided to revert creating the new unique index #8045 
Closing this one.

**Pull request closed without merging**